testing/matrix/linter/rules/modules/prometheus_rules.go

```diff
@@ -0,0 +1,162 @@
+/*
+Copyright 2022 Flant JSC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package modules
+
+import (
+	"crypto/sha256"
+	"os"
+	"os/exec"
+	"sync"
+
+	"sigs.k8s.io/yaml"
+
+	"github.com/deckhouse/deckhouse/testing/matrix/linter/rules/errors"
+	"github.com/deckhouse/deckhouse/testing/matrix/linter/storage"
+	"github.com/deckhouse/deckhouse/testing/matrix/linter/utils"
+)
+
+type checkResult struct {
+	success bool
+	errMsg  string
+}
+
+const promtoolPath = "/deckhouse/bin/promtool"
+
+var (
+	rulesCache   = map[string]checkResult{}
+	rulesCacheMu = sync.RWMutex{}
+)
+
+func promtoolAvailable() bool {
+	info, _ := os.Stat(promtoolPath)
```

Comment:
Why we suppress error there ?
from doc - Stat returns a FileInfo describing the named file.
If there is an error, it will be of type *PathError.

Reply:
I'm relying on `info` being nil in case of errors, checking it in following comparison in the return statement.
Is it considered a bad practice?

Reply:
Fixed error handling in df881a0d01d16cfceb2c26acbc0efe0e9ff955a3.